### PR TITLE
Warn instead of reporting an error when a webhook cannot be found

### LINKED ISSUE #25744: Unable to deliver webhook task[53215]: webhook does not exist [id: 134]
### Description

```
2023/07/07 15:54:26 ...s/webhook/webhook.go:125:handler() [E] Unable to deliver webhook task[53215]: webhook does not exist [id: 134]
```

### Gitea Version

v1.20+

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

./gitea web

### Database

MySQL

### PR DESCRIPTION
**Pull request by puni9869**
Attemp fix: #25744
Fixing the log level when we delete any repo then we get error hook not found by id. That should be warn level to reduce the noise in the logs.

### CODE REVIEW

**Comment by puni9869** [comment]
I am open for suggestions.

**Comment by lunny** [comment]
Maybe you need to check whether the error is record does not exist.

**Comment by puni9869** [comment]
> Maybe you need to check whether the error is record does not exist.

you mean error message == "record does not exist" ??

**Comment by lunny** [comment]
> > Maybe you need to check whether the error is record does not exist.
> 
> you mean error message == "record does not exist" ??

I mean if it's a `record doesn't exist` error, then create a warn log, otherwise still create an error log?

**Comment by puni9869** [comment]
image

But I have doubt 
image

There is a hook if Repo is deleted then we should post that event. But we are logging error.

Thoughts??

**Review by lunny — APPROVED** [review]

**Comment by puni9869** [comment]
Can I get one more approval on this PR.

File: services/webhook/webhook.go [hunk 1]
```diff
@@ -111,7 +111,11 @@ func handler(items ...int64) []int64 {
 	for _, taskID := range items {
 		task, err := webhook_model.GetHookTaskByID(ctx, taskID)
 		if err != nil {
-			log.Error("GetHookTaskByID[%d] failed: %v", taskID, err)
+			if webhook_model.IsErrHookTaskNotExist(err) {
```
Comment (delvh) [inline services/webhook/webhook.go]:
```suggestion
			if errors.Is(err, util.ErrNotExist) {
```

**Comment by puni9869** [comment]
+1 approval more to go

**Review by delvh — APPROVED** [review]

**Comment by delvh** [comment]
Backport?

**Comment by puni9869** [comment]
> Backport?

Yes

**Pull request merged**